src/constructs/lambda/lambda.ts

```diff
@@ -21,12 +22,28 @@ interface GuFunctionProps extends Omit<FunctionProps, "code"> {
   apis?: ApiProps[];
 }
 
+function defaultMemorySize(runtime: Runtime, memorySize?: number): number {
+  if (memorySize) {
+    return memorySize;
+  } else {
+    switch (runtime.family) {
+      case RuntimeFamily.JAVA:
+        return 1024;
+      default:
+        return 512;
+    }
+  }
+}
+
 export class GuLambdaFunction extends Function {
   constructor(scope: GuStack, id: string, props: GuFunctionProps) {
     const bucket = Bucket.fromBucketName(scope, `${id}-bucket`, props.code.bucket);
     const code = Code.fromBucket(bucket, props.code.key);
+    const { memorySize, timeout, ...rest } = props;
     super(scope, id, {
-      ...props,
+      ...rest,
+      memorySize: defaultMemorySize(props.runtime, memorySize),
+      timeout: props.timeout ? props.timeout : Duration.seconds(30),
```

Comment:
Minor but the following should achieve the same result with marginally less code
```suggestion
      timeout: props.timeout ?? Duration.seconds(30),
```